overlord/snapstate/policy/canremove_test.go

```diff
@@ -133,6 +144,19 @@ func (s *canRemoveSuite) TestOSInUseNotOK(c *check.C) {
 	c.Check(policy.NewOSPolicy("").CanRemove(s.st, snapst, snap.R(1), coreDev), check.Equals, policy.ErrInUseForBoot)
 }
 
+func (s *canRemoveSuite) TestOSNoSnapdNotOK(c *check.C) {
+	s.st.Lock()
+	defer s.st.Unlock()
+
+	snapstate.Set(s.st, "snapd", nil)
+
+	snapst := &snapstate.SnapState{
+		Current:  snap.R(1),
+		Sequence: snapstatetest.NewSequenceFromSnapSideInfos([]*snap.SideInfo{{Revision: snap.R(1), RealName: "core"}}),
+	}
+	c.Check(policy.NewOSPolicy("").CanRemove(s.st, snapst, snap.R(1), classicDev), check.Equals, policy.ErrSnapdNotInstalled)
```

Comment:
I presume that the idea here is to prevent snapd from downgrading by removing core from which snapd has re-executed. If so should we have a test that lets us remove snapd snap on systems that don't use this feature? If so perhaps a comment can be left for a follow-up that makes the intent clear.

Reply:
IIRC we always have either "snapd" or "core" regardless of reexec.

Reply:
we do because I think snap installs install snapd if neither is present